templates/eks-root-roles.yml

```diff
@@ -0,0 +1,237 @@
+---
+AWSTemplateFormatVersion: '2010-09-09'
+
+Parameters:
+  CodeBuildProjectName:
+    Type: String
+    Default: "crowdstrike-eks-codebuild"
+  CodeBuildRoleName:
+    Type: String
+    Default: "crowdstrike-eks-codebuild-role"
+  EventBridgeRoleName:
+    Type: String
+    Default: "crowdstrike-eks-eventbridge-role"
+  EKSExecutionRoleName:
+    Type: String
+    Default: "crowdstrike-eks-execution-role"
+  StagingS3Bucket:
+    Type: String
+  EventBusName:
+    Type: String
+    Default: "crowdstrike-eks-eventbus"
+  PermissionsBoundary:
+    Type: String
+    Description: The name of the policy used to set the permissions boundary for IAM roles.
+    Default: ''
+
+Conditions:
+  SetPermissionsBoundary: !Not [ !Equals [ !Ref PermissionsBoundary, '' ] ]
+
+Resources:
+  EKSEventBridgeRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Ref EventBridgeRoleName
+      PermissionsBoundary:
+        Fn::If:
+          - SetPermissionsBoundary
+          - !Sub 'arn:${AWS::Partition}:iam::${AWS::AccountId}:policy/${PermissionsBoundary}'
+          - Ref: AWS::NoValue
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service: events.amazonaws.com
+          Action: sts:AssumeRole
+      Policies:
+      - PolicyName: crowdstrike-eks-eb-role
+        PolicyDocument:
+          Version: 2012-10-17
+          Statement:
+          - Effect: Allow
+            Action: events:PutEvents
+            Resource: !Sub arn:${AWS::Partition}:events:${AWS::Region}:${AWS::AccountId}:event-bus/${EventBusName}
+
+  EKSExecutionRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      RoleName: !Ref EKSExecutionRoleName
```

Comment:
Consider not hardcoding the role name to handle the updates that require replacement of this resource.

Reply:
role names are parameterized to allow for cx who requires custom role naming convention